### ISSUE TITLE
Setting Host Name appers not to work in wifi router's website

### ISSUE THREAD

**Issue by VirtualGentlemen** [HomeSpan/HomeSpan#114]
Hi all,

I'd like to change the name of my ESP32's within my wifi network to distinguish wich are connected and wich are not. Currently they all have the name "esp32-arduino".

![InkedScreenshot 2021-07-17 111044_LI](https://user-images.githubusercontent.com/85340676/126032335-b8a533da-10bb-444a-83ef-ca5251e85b1e.jpg)

I tried changing DEFAULT_DISPLAY_NAME, DEFAULT_HOST_NAME and DEFAULT_MODEL_NAME but only DEFAULT_DISPLAY_NAME seems to have an affect in homekit device name.

`homeSpan.begin(Category::Thermostats, "asd", "asdf", "asdfg");`

Who's at fault? My Code, Homespan's code or is it not possible with my router?
Thanks in advance

_______________________________________________________________________________
Full setup code:
```
Serial.begin(115200);
  //wifi settings
  homeSpan.enableAutoStartAP();                               // enables auto start-up of Access Point when WiFi Credentials not found
  homeSpan.setApSSID("Homekit Enter Wifi Credentials");  // set AP SSID
  homeSpan.setApPassword("");                                 // set AP Password
  //io
  homeSpan.setControlPin(0); // set boot button
  homeSpan.setStatusPin(13); // sets Status Pin
  homeSpan.begin(Category::Thermostats, "asd", "asdf", "asdfg");
```

**Comment by HomeSpan** [comment]
I believe this is a router-specific issue and did some experimenting with my own setup (I use a Linksys Velop mesh network).  
First, I confirmed that the name that is used to identify the device in my Linksys app is not used for DNS purposes, and can be changed in the router app to anything I want (including names with spaces, special characters, etc.).  HomeSpan uses MDNS, so the name that appears in my router display does not mean much.

For a fresh ESP32 connecting the first time to my router, the router name matched the one HomeSpan broadcasts with MDNS.  However, the router seems to then permanently associate that name with the MAC Address of the ESP32.  If I change HomeSpan (as per your code) to broadcast a new name, this works for all DNS purposes, but Linksys continues to use the original name.  In order to get my router to recognize the new name, I tried changing the MAC address of the ESP32, and that seemed to do the trick (the router also issued the ESP32 a different IP Address).  

The other thing I experimented with is the `WiFi.setHostname(char *name)`.   This does not create an MDNS entry, but it does seem to be picked up by my router and will be used as the name.  If I specify a WiFi name that is different from the MDNS name, my router seems to choose the MDNS.  Also, the mapping of names to MAC Addresses seems to work the same way, so if I use `WiFi.setHostname(char *name)` to set the name in the router (by NOT using MDNS), this name stays fixed even if I reconnect using a different value for `name`.   The only way to get my router to recognize the new name is, as before, to change the MAC address of the ESP32.

Finally, I did note that if I connect without MDNS but include `WiFi.setHostname(char *name)` to create my own name on a fresh device, and then I add MDNS using a different name and re-connect, my router will adopt the MDNS name (as per the priority I found above).  But removing MDNS and reconnecting a third time does not result in any changes - once my router sees an MDNS name, it continues to use it as long as the MAC Address is the same.

In sum, routers seems to have their own logic to determine how to display device names.  Depending on which router you are using, its logic may be different than what I described above.  Seems unfortunate that router display names don't dynamically update and instead are hardcoded to the MAC Address (though maybe there is a good reason for this).

Thanks for raising this - I've run into this issue before trying to identify various HomePods and sometimes even different printers, but had not diagnosed the cause.
